packages/interface-adapter/lib/adapter/web3/index.ts

```diff
@@ -49,8 +49,13 @@ export class Web3InterfaceAdapter implements InterfaceAdapter {
     return this.web3.eth.getAccounts();
   }
 
-  public estimateGas(transactionConfig: Transaction) {
-    return this.web3.eth.estimateGas(transactionConfig);
+  public async estimateGas(transactionConfig: Transaction) {
+    try {
+      const gasEstimate = await this.web3.eth.estimateGas(transactionConfig);
+      return gasEstimate;
+    } catch {
```

Comment:
this `catch` should probably test to ensure that the `try` failed in the way your are expecting it to fail and only if it does return `null` - maybe the question is, "can `web3.eth.estimateGas` fail in a different way in which we would want to throw here?

Reply:
I made some changes to this because `contract-tests` do expect the VM error that throws here. So now we only catch the error if the user is debugging, so they can get past this point and debug to see why their code is reverting.